tools/bin/mtest

```diff
@@ -142,9 +141,16 @@ def subrun(cmd, test_dir=False, silent=False, ignore_failure=False):
     return (output, r.returncode)
 
 
+def sync():
+    # Ensure that e.g. config updates are sync'd before running make - have occasionally observed errors
+    # that might have come from make not seeing the changes to the config that have just been written.
+    subrun(["sync", ROOT], silent=True)
```

Comment:
I have seen some occasional intermittent issues where doing `mtest; programs/test/benchmark` failed unexpectedly, perhaps because things were out of sync. So this is a bit of a guess.

This is quite possibly not the right fix; do you have a better suggestion?

Reply:
What does "failed unexpectedly" mean? What error?

This sounds most like a parallel job was running and hadn't finished - i.e. that `mtest` finished before all the compilations.

And I suspect the reason adding `sync` works is because there would be a delay, so it's the equivalent of doing `mtest; sleep 1; programs/test/benchmark`, which clearly isn't a good fix.

Reply:
Indeed. But `mtest` just runs `make`, and waits for it to exit, so not much we can do here? I'll remove the sync for now.